### ISSUE TITLE
when loading the generated Gtfs, an error is reported for checkheaders.

### ISSUE THREAD

**Issue by pansy0425** [graphhopper/graphhopper#2509]
Hello, I'm studying the public transit module in graphoper recently.

When I start the service for the first time (run graphhopperapplication.java), cache data is generated in directory( graphs/brandenburg-with-transit ), which contains two parts(osm related-[properties, node, edges, string_index_vals, geometry, string_index_keys], and gtfs related-[transit_shcedule,transit_schedule.p,gtfs_0,gtfs_0.p]). At this time, the PtRouteResource.java in the reader-gtfs module can call route() interface to get bus travel paths, and the results returned by the incoming parameters also contain bus travel path information, which shows that the graphoper I built is successful and the generated graph in directory( graphs/brandenburg-with-transit ) is OK.

But when I stop the service and want to load the service for the second time with the already existed graph, which is in directory ( graphs/brandenburg-with-transit ). At this time, the code reads the data related to graphs from this directory ( graphs/brandenburg-with-transit ), but an error is reported:
java.io.IOError: java.io.IOException: Wrong index checksum, store was not closed properly and could be corrupted.
	at org.mapdb.StoreDirect.checkHeaders(StoreDirect.java:269)
	at org.mapdb.StoreDirect.<init>(StoreDirect.java:207)
	at org.mapdb.DBMaker.extendStoreDirect(DBMaker.java:971)
	at org.mapdb.DBMaker.makeEngine(DBMaker.java:758)
	at org.mapdb.DBMaker.make(DBMaker.java:701)
	at com.graphhopper.reader.gtfs.GtfsStorage.loadExisting(GtfsStorage.java:125)
	at com.graphhopper.reader.gtfs.GraphHopperGtfs.importPublicTransit(GraphHopperGtfs.java:131)
	at com.graphhopper.GraphHopper.postProcessing(GraphHopper.java:973)
	at com.graphhopper.GraphHopper.load(GraphHopper.java:803)
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:638)
	at com.graphhopper.http.GraphHopperManaged.start(GraphHopperManaged.java:125)
	at io.dropwizard.lifecycle.JettyManaged.doStart(JettyManaged.java:27)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.server.Server.start(Server.java:407)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.server.Server.doStart(Server.java:371)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at io.dropwizard.cli.ServerCommand.run(ServerCommand.java:53)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:45)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:87)
	at io.dropwizard.cli.Cli.run(Cli.java:78)
	at io.dropwizard.Application.run(Application.java:94)
	at com.graphhopper.http.GraphHopperApplication.main(GraphHopperApplication.java:36)
Caused by: java.io.IOException: Wrong index checksum, store was not closed properly and could be corrupted.
	... 25 more

How to solve this problem? I find the solution to online query is to let me delete the graph Location folder, and then the program will regenerate the new graph Location folder.
(https://discuss.graphhopper.com/t/java-io-ioerror-java-io-ioexception-wrong-index-checksum-store-was-not-closed-properly-and-could-be-corrupted/4908)

But my subsequent requirements want to generate graph only once，from loading original data, which is easy to use the generated graph in directory( graphs/brandenburg-with-transit ) all the time, so that I don’t need to spend a lot of time loading from the original file.

So I want to ask: how can I solve the problem of error reporting without deleting the previously generated graph in directory( graphs/brandenburg-with-transit )? 
(that is, when loading the generated Gtfs, an error is reported for checkheaders.)

**Comment by karussell** [comment]
How did you stop the service? When I do Ctrl+C in the command line I cannot reproduce this problem the second time it starts in 14s instead of 241s.

Are you on master branch or the stable `4.x`?

**Comment by michaz** [comment]
Yes, I think the process was KILLed here rather than stopped. Mapdb doesn't like that a lot.

For now I would call this "won't fix" -- processes should be terminated cleanly, and this isn't live data, it's just a preprocessed GTFS, so I think this is acceptable.

**Comment by pansy0425** [comment]
> Yes, you are right. MapDB cannot be killed directly, but needs to be closed.

Therefore, my solution is: when creating mapdb, add “. transactiondisable (). closeonvmshutdown ()” to solve the possibility that mapdb will be destroyed when shutting down jvm.

Now I did the same operations as before (When I start the service for the first time (run graphhopperapplication.java), cache data is generated in directory( graphs/brandenburg-with-transit ), which contains two parts(osm related-[properties, node, edges, string_index_vals, geometry, string_index_keys], and gtfs related-[transit_shcedule,transit_schedule.p,gtfs_0,gtfs_0.p]). At this time, the PtRouteResource.java in the reader-gtfs module can call route() interface to get bus travel paths, and the results returned by the incoming parameters also contain bus travel path information, which shows that the graphoper I built is successful and the generated graph in directory( graphs/brandenburg-with-transit ) is OK.)
Then I stop the GraphHopperApplication in the Ideal tool and want to load the service for the second time with the already existed graph, which is in directory ( graphs/brandenburg-with-transit ). At this time, the code reads the data related to graphs from this directory ( graphs/brandenburg-with-transit ) successfully!

Now traveling based on bus routes(the Public Transit module), the generated road network can be directly used, and the data loading speed is greatly improved!

**Comment by pansy0425** [comment]
> How did you stop the service? When I do Ctrl+C in the command line I cannot reproduce this problem the second time it starts in 14s instead of 241s.
> 
> Are you on master branch or the stable `4.x`?

I already solve the problem. Thanks!
